codegen/src/main/java/software/amazon/awssdk/codegen/IntermediateModelBuilder.java

```diff
@@ -193,21 +189,50 @@ private void linkCustomAuthorizationToRequestShapes(IntermediateModel model) {
                      throw new RuntimeException(String.format("Operation %s has unknown input shape",
                                                               operation.getOperationName()));
                  }
-                 if (AuthType.CUSTOM.equals(c2jOperation.getAuthType())) {
-                     AuthorizerModel auth = model.getCustomAuthorizers().get(c2jOperation.getAuthorizer());
-                     if (auth == null) {
-                         throw new RuntimeException(String.format("Required custom auth not defined: %s",
-                                                                  c2jOperation.getAuthorizer()));
-                     }
-                     shape.setRequestSignerClassFqcn(model.getMetadata().getAuthPolicyPackageName() + '.' +
-                                                     auth.getInterfaceName());
-                 } else if (AuthType.IAM.equals(c2jOperation.getAuthType())) {
-                     model.getMetadata().setRequiresIamSigners(true);
-                     shape.setRequestSignerClassFqcn("software.amazon.awssdk.opensdk.protect.auth.IamRequestSigner");
+
+                 if (model.getMetadata().getProtocol() == Protocol.API_GATEWAY) {
+                     linkCustomAuthorizationToRequestShapeForApiGatewayProtocol(model, c2jOperation, shape);
+                 } else {
+                     linkCustomAuthorizationToRequestShapeForAwsProtocol(c2jOperation.getAuthType(), shape);
                  }
              });
     }
 
+    private void linkCustomAuthorizationToRequestShapeForApiGatewayProtocol(IntermediateModel model,
+                                                                            Operation c2jOperation,
+                                                                            ShapeModel shape) {
+        if (AuthType.CUSTOM.equals(c2jOperation.getAuthType())) {
+            AuthorizerModel auth = model.getCustomAuthorizers().get(c2jOperation.getAuthorizer());
+            if (auth == null) {
+                throw new RuntimeException(String.format("Required custom auth not defined: %s",
+                                                         c2jOperation.getAuthorizer()));
+            }
+            shape.setRequestSignerClassFqcn(model.getMetadata().getAuthPolicyPackageName() + '.' +
+                                            auth.getInterfaceName());
+        } else if (AuthType.IAM.equals(c2jOperation.getAuthType())) {
+            model.getMetadata().setRequiresIamSigners(true);
+            // TODO IamRequestSigner does not exist
+            shape.setRequestSignerClassFqcn("software.amazon.awssdk.opensdk.protect.auth.IamRequestSigner");
+        }
+    }
+
+    private void linkCustomAuthorizationToRequestShapeForAwsProtocol(AuthType authType, ShapeModel shape) {
```

Comment:
I wouldn't call this "custom" so much as operation specific.

Reply:
Used the same names from v1. Removed "Custom" from method name. Btw these are private methods, so not worried about naming as long as they are understandable.